tools/src/commands/AndroidNativeUnitTests.ts

```diff
@@ -52,23 +52,17 @@ export async function androidNativeUnitTests({
   const packageNamesFilter = packages ? packages.split(',') : [];
 
   const androidPackages = await filterAsync(allPackages, async (pkg) => {
-    const pkgSlug = pkg.packageSlug;
-
     if (packageNamesFilter.length > 0 && !packageNamesFilter.includes(pkg.packageName)) {
       return false;
     }
 
     let includesTests;
-    if (type === 'instrumented') {
-      includesTests =
-        pkg.isSupportedOnPlatform('android') &&
-        (await pkg.hasNativeInstrumentationTestsAsync('android')) &&
-        !excludedInTests.includes(pkgSlug);
-    } else {
-      includesTests =
-        pkg.isSupportedOnPlatform('android') &&
-        (await pkg.hasNativeTestsAsync('android')) &&
-        !excludedInTests.includes(pkgSlug);
+    if (pkg.isSupportedOnPlatform('android') && !excludedInTests.includes(pkg.packageSlug)) {
+      if (type === 'instrumented') {
+        includesTests = await pkg.hasNativeInstrumentationTestsAsync('android');
+      } else {
+        includesTests = await pkg.hasNativeTestsAsync('android');
+      }
```

Comment:
Whenever you need to use `let` instead of `const`, it's nicer to extract this to the separate function so you can use `const`, but I'm fine with that too.

Reply:
Sure, can follow up with PR for that, but I think code is still a bit less convoluted than original.